snuba_sdk/mql/mql.py

```diff
@@ -53,21 +53,25 @@
 variable = "$" ~r"[a-zA-Z0-9_.]+"
 nested_expression = open_paren _ expression _ close_paren
 
-function = (aggregate / curried_aggregate / arbitrary_function) (group_by)?
-aggregate = aggregate_name (open_paren _ filter _ close_paren)
-curried_aggregate = curried_aggregate_name (open_paren _ aggregate_list? _ close_paren) (open_paren _ filter _ close_paren)
-arbitrary_function = arbitrary_function_name (open_paren ( _ filter _ ) (_ comma _ param_expression)* close_paren)
+function = (curried_aggregate / curried_arbitrary_function / aggregate / arbitrary_function) (group_by)?
+aggregate = aggregate_name (open_paren _ inner_filter _ close_paren)
+aggregate_name = ~r"[a-zA-Z0-9_]+"
+arbitrary_function = arbitrary_function_name (open_paren ( _ expression _ ) (_ comma _ expression)* close_paren)
+arbitrary_function_name = ~r"[a-zA-Z0-9_]+"
+curried_aggregate = curried_aggregate_name (open_paren _ aggregate_list? _ close_paren) (open_paren _ inner_filter _ close_paren)
+curried_aggregate_name = "quantiles" / "histogram"
```

Comment:
Why is this still hardcoded? If we can, I would treat this the same way as regular aggregates.

Reply:
Are you referring to the `curried_aggregate`? We can treat them as `curried_arbitrary_function`.

Reply:
Removed `curried_aggregate`.

Reply:
Updated to allow arbitrary curried names.